Update Readme.md and Makefile to new syntax. Upgrade pre-commit hooks to v0.1.4

soerenmartius opened the pull request.

mariux commented on README.md:
```diff
@@ -31,21 +42,14 @@ In contrast to the plain `aws_iam_role` resource this module simplifies adding I
   Create an inline IAM policy, Attach custom or AWS managed policies. Create an IAM instance profile.
 
 ## Getting Started
-Basic usage for granting an AWS Account with Account ID `123456789012` access to assume a role that grants
-full access to AWS Simple Storage Service (S3)
+
+Basic usage:
```
please do not change already written documentation back to the template

soerenmartius replied:
resolved in d93f36f

mariux commented on README.md:
````diff
@@ -31,21 +42,14 @@ In contrast to the plain `aws_iam_role` resource this module simplifies adding I
+Basic usage:
 
 ```hcl
 module "role-s3-full-access" {
   source = "git@github.com:mineiros-io/terraform-aws-iam-role.git?ref=v0.0.1"
 
-  name = "S3FullAccess"
-
-  assume_role_principals = [
-    {
-      type        = "AWS"
-      identifiers = ["arn:aws:iam::123456789012:root"]
-    }
-  ]
+  name = "s3-full-access"
 
   policy_statements = [
````
revert

soerenmartius replied:
resolved in d93f36f

soerenmartius commented:
> Please revert redundant links to links with same http markdown handles pure links just fine by default..

I would rather stick to GH flavored markdown: https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md034---bare-url-used
https://github.github.com/gfm/#links

mariux commented:
> > Please revert redundant links to links with same http markdown handles pure links just fine by default..
> 
> I would rather stick to GH flavored markdown: https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md034---bare-url-used
> https://github.github.com/gfm/#links

See "Automatic linking for URLs" in https://guides.github.com/features/mastering-markdown/ official github guide. you reference to the implementation details when writing github-md parsers... we are users..  but for your reference see also https://github.github.com/gfm/#example-619 ;)

soerenmartius commented:
> > > Please revert redundant links to links with same http markdown handles pure links just fine by default..
> > 
> > 
> > I would rather stick to GH flavored markdown: https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md034---bare-url-used
> > https://github.github.com/gfm/#links
> 
> See "Automatic linking for URLs" in https://guides.github.com/features/mastering-markdown/ official github guide. you reference to the implementation details when writing github-md parsers... we are users.. but for your reference see also https://github.github.com/gfm/#example-619 ;)

resolved in 08a5c95

mariux commented on README.md:
```diff
@@ -222,17 +245,20 @@ for details and use-cases.
-  - Roles: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html
-  - Policies: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies.html
-  - Instance Profile: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html
+  - Roles: [https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html)
+  - Policies: [https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies.html](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies.html)
+  - Instance Profile: [https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html)
```
please remove redundant link naming

soerenmartius replied:
resolved in 4ab8a34

soerenmartius commented:
> lgtm..
> 
> please clean git history creating at least two different commits for build-system updates and documentation updates as both are not related...
> 
> and rename the PR accordingly as the title describes a completely different scope.
> 
> thx ;)

Cleaned up the history so we have two commits only. Changed the title of this PR also.
